tiledb/sm/fragment/single_fragment_info.h

```diff
@@ -273,6 +280,8 @@ class SingleFragmentInfo {
    * domain is the same as the non-empty domain.
    */
   NDRange expanded_non_empty_domain_;
+  /** The fragment's mbrs. */
+  std::vector<NDRange> mbrs_;
```

Comment:
See my other comments about lazy fetching of R-trees. The R-tree is large fragments will be large. I don't think we should be storing it twice, although I am up for discussion. Ideally they should be retrieved directly from the R-tree to be loaded in the `FragmentMetadata` instance this `SingleFragmentInfo` instance corresponds to.

Reply:
To address this, we will need to store a pointer to the MBRs in the *leaves* of the R-tree of the  `FragmentMetadata` class.

Reply:
OK, I see. I'll make the modification so that it doesn't call `load_rtree()` every time

Reply:
`load_rtree` is a noop when the tree has been already loaded. The issue here is that we should load the R-tree lazily only when the new MBR-related APIs are called.